railties/lib/rails/application/configuration.rb

```diff
@@ -512,7 +512,7 @@ def colorize_logging=(val)
 
       def secret_key_base
         @secret_key_base || begin
-          self.secret_key_base = if Rails.env.local? || ENV["SECRET_KEY_BASE_DUMMY"]
+          self.secret_key_base = if generate_local_secret?
```

Comment:
If `secret_key_base` does nothing when `generate_local_secret?` could this do the following instead?
```
    def secret_key_base
       @secret_key_base || begin
          if generate_local_secret?
            generate_local_secret
          else
            self.secret_key_base = ENV["SECRET_KEY_BASE"] || Rails.application.credentials.secret_key_base
          end
       end
    end
```